pycbc/waveform/generator.py

```diff
@@ -1198,57 +1199,57 @@ def generate(self, **kwargs):
 #
 
 
-def select_waveform_generator(approximant):
-    """Returns the single-IFO generator for the approximant.
-
-    Parameters
-    ----------
-    approximant : str
-        Name of waveform approximant. Valid names can be found using
-        ``pycbc.waveform`` methods.
-
-    Returns
-    -------
-    generator : (PyCBC generator instance)
-        A waveform generator object.
-
-    Examples
-    --------
-    Get a list of available approximants:
-    >>> from pycbc import waveform
-    >>> waveform.fd_approximants()
-    >>> waveform.td_approximants()
-    >>> from pycbc.waveform import ringdown
-    >>> ringdown.ringdown_fd_approximants.keys()
-
-    Get generator object:
-    >>> from pycbc.waveform.generator import select_waveform_generator
-    >>> select_waveform_generator(waveform.fd_approximants()[0])
-    """
-    # check if frequency-domain CBC waveform
+def select_waveform_generator(approximant, domain):
+    """Returns the single-IFO generator for the approximant."""
+
+    generator_map = {
+        'fd': {
+            'cbc': FDomainCBCGenerator,
+            'ringdown': {
+                'FdQNMfromFinalMassSpin': FDomainMassSpinRingdownGenerator,
+                'FdQNMfromFreqTau': FDomainFreqTauRingdownGenerator,
+            }
+        },
+        'td': {
+            'cbc': TDomainCBCGenerator,
+            'ringdown': {
+                'TdQNMfromFinalMassSpin': TDomainMassSpinRingdownGenerator,
+                'TdQNMfromFreqTau': TDomainFreqTauRingdownGenerator,
+            },
+            'supernovae': {
+                'CoreCollapseBounce': TDomainSupernovaeGenerator
+            }
+        }
+    }
+
+    gen = None
```

Comment:
Add a check here that domain is set to one of the recognized values (`None`, `'td'`, or `'fd'`). If not, raise a ValueError.

Reply:
@kkacanja If there are good reason, one can sometimes bend / ignore the CC recommendations. One shouldn't do that by default, but we shouldn't hold so strictly to them that we make it impossible to write sensible code.